### ISSUE TITLE
Cannot Install Brave: "The download failed," Keeps Failing

### ISSUE THREAD

**Issue by tominterval** [brave/brave-browser#18300]
## Description 
On an Asus PC running Windows 8.1, I cannot download Brave for the first time because "The download failed" and keeps failing (see two screencaps).

## Steps to Reproduce

   1. Double-click on the installation file (BraveBrowserSetup.exe).
   2. Click "Yes" on the User Account Control pop-up.

## Actual result:
A box pops up with blue status bar with message: "Egads! Download failed. Retrying... X second(s) remaining" (screencap attached). After the status box goes away, another box pops up with "The download failed." Neither box has an error number.

## Expected result:
That Brave downloads and installs.

## Reproduces how often: 
Easily reproduced over the last two days.

## Desktop Brave version:
The latest version available for 64-bit as of yesterday on brave.com.

## Device details:
Asus PC running on Windows 8.1. Used Chrome browser to download executable install file.

## Other Additional Information:
To clarify, this is the first time I'm trying to download Brave. Please see two attached screencaps.

![downloadfailed01](https://user-images.githubusercontent.com/58096621/134436302-c7042036-1ed0-4fc4-8ec1-2c898cd52309.jpg)
![downloadfailed02](https://user-images.githubusercontent.com/58096621/134436304-69868963-982c-421d-97f1-9bdc10ed9faf.jpg)

## Miscellaneous Information:
30.3GB free space on C: drive. No other apps running. No virus scanners scanning, no lords a leaping.

**Comment by Nihal247** [comment]
@tominterval, if you're continuously facing this problem, would you like to try installing Brave through a package manager, like Chocolatey.

Steps:

1. Open a PowerShell (Admin) Window.
2. Copy this full command and paste it in the PowerShell (Admin) Window to install chocolatey 👇
`Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))`
3. After installation, type `choco install brave`, this installs Brave.
4. Type `y`  for confirming the installation.
5. Wait for the installation process to get completed.
6. After installation, search for 'Brave', and you should see it.
7. To uninstall Chocolatey, open File Explorer, navigate to `C:\ProgramData`, and delete the `chocolatey` folder.

Hope that helps!

Thanks!

**Comment by tominterval** [comment]
> I wasn't able to reproduce this but @tominterval, if you're continuously facing this problem, would you like to try installing Brave through a package manager, like Chocolatey?
> 
> Steps:
> 
> 1. Open a PowerShell (Admin) Window.
> 2. Copy this full command and paste it in the PowerShell (Admin) Window to install chocolatey point_down
>    `Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))`
> 3. After installation, type `choco install brave`, this installs Brave.
> 4. Type `y`  for confirming the installation.
> 5. Wait for the installation process to get completed.
> 6. After installation, search for 'Brave', and you should see it.
> 7. To uninstall Chocolatey, open File Explorer, navigate to `C:\ProgramData`, and delete the `chocolatey` folder.
> 
> Or
> 
> See if Brave Beta or Brave Nightly gets installed through the .exe file [here](https://github.com/brave/brave-browser/releases) (remember these can be unstable, and try to perform the chocolatey process first).
> 
> Hope that helps!
> 
> Thanks!

Hi Nihal247. Thanks so much for the suggestions. So there's no way to get the executable file working from the Brave website? I'm not familiar with package managers. Thanks!

**Comment by Nihal247** [comment]
@tominterval, kindly check if you have a 32-bit PC, if it so, download the 32-bit one, if not, are you running the installation with administrative privileges? If it's failing, try installing Brave without administrative privileges, it might help.

**Comment by tominterval** [comment]
> @tominterval, kindly check if you have a 32-bit PC, if so, download the 32-bit one, if not, are you running the installation with administrative privileges? If it's failing, try installing Brave without administrative privileges, it might help.

Hi Nihal247. Thanks for the suggestions. I'm on 64-bit. No matter what I do or how I try to install it (admin privileges or not), it's not working. I've never seen this happen with any other installation file for any program in the universe. This has to be happening to other people, yet no one seems to be able to resolve it other than workarounds that, frankly, don't make any sense to me. One should be able to download an executable installation file, double-click on it, approve (yes or no), and the install the software. Not so with Brave. It really IS private—SO private that not everyone can even install it. Sorry, but WTF?

**Comment by tominterval** [comment]
P.S. Do Brave developers ever look at this forum?

> > I wasn't able to reproduce this but @tominterval, if you're continuously facing this problem, would you like to try installing Brave through a package manager, like Chocolatey?
> > Steps:
> > 
> > 1. Open a PowerShell (Admin) Window.
> > 2. Copy this full command and paste it in the PowerShell (Admin) Window to install chocolatey point_down
> >    `Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))`
> > 3. After installation, type `choco install brave`, this installs Brave.
> > 4. Type `y`  for confirming the installation.
> > 5. Wait for the installation process to get completed.
> > 6. After installation, search for 'Brave', and you should see it.
> > 7. To uninstall Chocolatey, open File Explorer, navigate to `C:\ProgramData`, and delete the `chocolatey` folder.
> > 
> > Or
> > See if Brave Beta or Brave Nightly gets installed through the .exe file [here](https://github.com/brave/brave-browser/releases) (remember these can be unstable, and try to perform the chocolatey process first).
> > Hope that helps!
> > Thanks!
> 
> Hi Nihal247. Thanks so much for the suggestions. So there's no way to get the executable file working from the Brave website? I'm not familiar with package managers. Thanks!

Also, thank you so much for taking the time to reply. The only reason I didn't try the Chocolately workaroud you suggested is because it's totally unfamiliar to me, and I don't want to "break" anything on my PC, if that makes any sense. I'm just confused as to why the installation file doesn't work. Isn't that kind of a big bug? I couldn't' be the only person who can't install Brave.

**Comment by bsclifton** [comment]
Hi @tominterval - what you are experiencing hasn't been reported before (this is NOT a known issue). Thanks for reporting 😄 

Do you have anything being blocked either by your hosts file? Windows Firewall? or on your home network / router? The installer distributed by brave.com simply hits our update servers to download the actual browser itself. This download is what is causing the failure. `eventvwr` (on your machine) may contain more information

cc: @brave/devops-team for tips on troubleshooting Omaha (the updater)I forget if we log Omaha by default? or if we need to set registry). I have everything logging for me at `C:\Omaha log.txt`. Opening this up gives some clues

You can try downloading the full installer here (won't reach out to internet):
https://github.com/brave/brave-browser/releases/tag/v1.29.81

If you do want to try out last known good release (from that link), you'll want to pick the following executable:
https://github.com/brave/brave-browser/releases/download/v1.29.81/BraveBrowserStandaloneSetup.exe

**Comment by mihaiplesa** [comment]
@bsclifton logging can be turned on as in https://community.brave.com/t/error-with-brave-update/38909

**Comment by bsclifton** [comment]
UPDATE: this may be unintentionally caused by https://github.com/brave/devops/pull/5870 (link is private to employees)

We have several reports here:
- https://community.brave.com/t/download-failed-in-installer/285718/3
- https://community.brave.com/t/egads-downlod-failed-brave-pc/285720
- https://community.brave.com/t/i-cant-download-the-browser/285569
- https://community.brave.com/t/i-cant-download-brave/285480/4

**Comment by bsclifton** [comment]
Working on a solution here - thanks @justnom 😄 Will share here (and on community) when we have a potential fix

**Comment by bsclifton** [comment]
OK folks - this should be resolved. @tominterval were you able to try again?

**Comment by bsclifton** [comment]
Closing as this should be resolved with https://github.com/brave/devops/pull/5912 (internal only issue; as this deals with hosting / infrastructure)

Please reopen or reply here with feedback 😄 If you did experience the issue - how is it now? Working? Not working? Let us know!

**Comment by tominterval** [comment]
> OK folks - this should be resolved. @tominterval were you able to try again?

Hi Brian! Yes, it worked! Thank you all so much for your time and help.

**Comment by tominterval** [comment]
> UPDATE: this may be unintentionally caused by https://github.com/brave/devops/pull/5870 (link is private to employees)
> 
> We have several reports here which I'm following up on:
> 
> * https://community.brave.com/t/download-failed-in-installer/285718/3
> * https://community.brave.com/t/egads-downlod-failed-brave-pc/285720
> * https://community.brave.com/t/i-cant-download-the-browser/285569
> * https://community.brave.com/t/i-cant-download-brave/285480/4

Nice work, man. It installed! Thank you again. Super appreciate it and look forward to trying out this browser.

**Comment by jabowery** [comment]
It just happened to me.  Since there is no version information provided for the Setup executable, here is its length:

(1,255,336 bytes)

That executable obtained from:

https://brave.com/download/

Running on:

Device name	jabowery-PC
Processor	AMD Phenom(tm) II X6 1055T Processor   2.80 GHz
Installed RAM	16.0 GB
Device ID	432451C9-1CC8-4C97-8249-D65FC500A5FA
Product ID	00330-80000-00000-AA955
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

**Comment by Acroxis** [comment]
This is still happening to me in April 2022!  I had to reinstall Brave after completely removing it and now I also can't down;oad the release version from Brave.

The exact same thing is happening in post 1 with the same screenshots.  I spent half a day and gave up - it's impossible for me to locate a stable release 1.37.116 (?).  The only build I've got is a nightly standalone that I don't want.  I'm sick of Brave updating while I'm using it and crashing websites and slowing everything down

What is the actual solution to this issue?  It's obviously an issue with your servers.  The bravesetup.exe file downloads, but the installation after launcing the exe fails with the same screenshots as post 1.  I can't believe this is still happening after all this time.

I'm on a Windows 10 64 bit version 21H1 (19043.1526)
